### PR TITLE
New Proposal Modal

### PR DESCRIPTION
**Pull request by cdt-eth**
Updates:
- proposal page is now a modal to easily cycle through each
- navigate between props using L/R arrow keys
- overflow content gradient & "More" scroll button
- Wallet Connect & Create Proposal buttons via prop page
- vote within modal
- modal to see current pre-submission vote allotment
- banner to denote winners

### CODE REVIEW

**Review by cryptoseneca — CHANGES_REQUESTED** [review]
Great job all around, the modal is a great UX improvement. Some UX stuff I noticed:

- “tldr” and “description” titles should have diff font color to differentiate from actual prop content 
- “More” button when prop modal content is long is very hard to read - let’s add shadow (or something) to make it stand out more

on design side: 
- let's add "connect to vote" on prop modal if in round_voting state && not connected

other:
- I noticed a few components that made it to the `pages` directory that should not be there, let's move those [1]


[1]
Screen Shot 2022-11-18 at 11 02 25 AM

**Comment by cdt-eth** [comment]
> * I noticed a few components that made it to the `pages` directory that should not be there, let's move those [1]

addressed https://github.com/Prop-House/prop-house-monorepo/pull/299/commits/d81751c82a709d2bdf3d254f6121e83b840c19bf

File: packages/prop-house-webapp/src/components/ProposalFooter/index.tsx [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+import { VoteAllotment } from '../../types/VoteAllotment';
+import { voteWeightForAllottedVotes } from '../../utils/voteWeightForAllottedVotes';
+import { Dispatch, SetStateAction } from 'react';
+import { ImArrowLeft2, ImArrowRight2 } from 'react-icons/im';
+import { Direction } from '@nouns/prop-house-wrapper/dist/builders';
+import Divider from '../Divider';
+import { AuctionStatus, auctionStatus } from '../../utils/auctionStatus';
+import { useEthers } from '@usedapp/core';
+
+const ProposalFooter: React.FC<{
```
[the remark this answers is not in the document]
Comment (cdt-eth) [reply packages/prop-house-webapp/src/components/ProposalFooter/index.tsx]: https://github.com/Prop-House/prop-house-monorepo/pull/299/commits/98db1a496f4bd411c2f9766995aafc10a3b80c7c

File: packages/prop-house-webapp/src/components/ProposalFooter/index.tsx [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+  round: StoredAuction;
```
[the remark this answers is not in the document]
Comment (cdt-eth) [reply packages/prop-house-webapp/src/components/ProposalFooter/index.tsx]: https://github.com/Prop-House/prop-house-monorepo/pull/299/commits/c6cfca1cbbaec6564c45a3b23524e3b57cdfd870

File: packages/prop-house-webapp/src/components/ProposalModal/index.tsx [hunk 3]
```diff
@@ -0,0 +1,325 @@
...
+      >
+        {proposal &&
+        proposals &&
+        proposals.length > 0 &&
+        currentPropIndex &&
+        currentProposal &&
+        round ? (
+          <>
+            <ProposalHeaderAndBody
+              proposal={proposal}
```
[the remark this answers is not in the document]
Comment (cdt-eth) [reply packages/prop-house-webapp/src/components/ProposalModal/index.tsx]: https://github.com/Prop-House/prop-house-monorepo/pull/299/commits/c6cfca1cbbaec6564c45a3b23524e3b57cdfd870

File: packages/prop-house-webapp/src/components/ProposalModal/index.tsx [hunk 4]
```diff
@@ -0,0 +1,325 @@
...
+              currentProposal={currentProposal}
+              currentPropIndex={currentPropIndex}
+              handleDirectionalArrowClick={handleDirectionalArrowClick}
+              handleClosePropModal={handleClosePropModal}
+              isWinner={winningIds && isWinner(winningIds, proposal.id)}
+              hideScrollButton={hideScrollButton}
+              setHideScrollButton={setHideScrollButton}
+            />
+
+            <ProposalFooter
```
[the remark this answers is not in the document]
Comment (cdt-eth) [reply packages/prop-house-webapp/src/components/ProposalModal/index.tsx]: https://github.com/Prop-House/prop-house-monorepo/pull/299/commits/c6cfca1cbbaec6564c45a3b23524e3b57cdfd870

File: packages/prop-house-webapp/src/components/VoteConfirmationModal/index.tsx [hunk 5]
```diff
@@ -30,7 +30,11 @@ const VoteConfirmationModal: React.FC<{
     <Modal isOpen={showNewModal} onRequestClose={closeModal} className={clsx(classes.modal)}>
       <div className={classes.titleContainer}>
         <p className={classes.modalTitle}>
-          Cast {voteAllotments.reduce((total, prop) => (total = total + prop.votes), 0)} votes?
+          Cast {voteAllotments.reduce((total, prop) => (total = total + prop.votes), 0)}{' '}
```
[the remark this answers is not in the document]
Comment (cdt-eth) [reply packages/prop-house-webapp/src/components/VoteConfirmationModal/index.tsx]: https://github.com/Prop-House/prop-house-monorepo/pull/299/commits/3fb6adb26511cba31d51391497dbb18513ade0ac

File: packages/prop-house-webapp/src/components/ProposalModal/index.tsx [hunk 6]
```diff
@@ -0,0 +1,325 @@
...
+        id="propModal"
+      >
+        {proposal &&
+        proposals &&
+        proposals.length > 0 &&
+        currentPropIndex &&
+        currentProposal &&
+        round ? (
+          <>
+            <ProposalHeaderAndBody
```
[the remark this answers is not in the document]
Comment (cdt-eth) [reply packages/prop-house-webapp/src/components/ProposalModal/index.tsx:284]: So we already do, @cryptoseneca. The `ProposalHeader` is [here](https://github.com/Prop-House/prop-house-monorepo/blob/3fb6adb26511cba31d51391497dbb18513ade0ac/packages/prop-house-webapp/src/components/ProposalHeaderAndBody/ProposalHeaderAndBody.tsx#L77-L91) and the body is the `ProposalContent` component, [on line 106](https://github.com/Prop-House/prop-house-monorepo/blob/3fb6adb26511cba31d51391497dbb18513ade0ac/packages/prop-house-webapp/src/components/ProposalHeaderAndBody/ProposalHeaderAndBody.tsx#L106), which is used in several other places in the codebase. 

The entire `ProposalBodyAndHeader` component is to encompass all the overflow scrolling logic & UI and keep it out of its parent, `ProposalModal`, which is already busy enough.

File: packages/prop-house-webapp/src/components/ProposalModal/index.tsx [hunk 7]
```diff
@@ -0,0 +1,325 @@
...
+    };
+
+    fetchAuctionProposals();
+    return () => {
+      dispatch(setActiveProposals([]));
+    };
+  }, [currentProposal, dispatch, round]);
+
+  const handleDirectionalArrowClick = (direction: Direction) => {
+    if (!community || !round) return;
```
[the remark this answers is not in the document]
Comment (cdt-eth) [reply packages/prop-house-webapp/src/components/ProposalModal/index.tsx]: https://github.com/Prop-House/prop-house-monorepo/pull/299/commits/b1074d6841b9d24a21048a9569f395733e077400

File: packages/prop-house-webapp/src/components/ProposalHeader/index.tsx [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+  proposalId: number;
+  backButton: React.ReactNode;
+  propIndex: number | undefined;
+  numberOfProps: number;
+  handleDirectionalArrowClick: (e: any) => void;
+  isFirstProp: boolean;
+  isLastProp: boolean;
+}
+
+const ProposalHeader: React.FC<ProposalHeaderProps> = props => {
```
[the remark this answers is not in the document]
Comment (cdt-eth) [reply packages/prop-house-webapp/src/components/ProposalHeader/index.tsx]: https://github.com/Prop-House/prop-house-monorepo/pull/299/commits/d05a26f3b6bf841b3417565d7e2378f5ea95aa1a

File: packages/prop-house-webapp/src/components/ProposalModal/index.tsx [hunk 9]
```diff
@@ -0,0 +1,325 @@
...
+      setShowErrorModal(true);
+    }
+  };
+
+  const winningIds = round && getWinningIds(proposals, round);
+
+  const [hideScrollButton, setHideScrollButton] = useState(false);
+
+  const modal = document.querySelector('#propModal');
+  useEffect(() => {
```
[the remark this answers is not in the document]
Comment (cdt-eth) [reply packages/prop-house-webapp/src/components/ProposalModal/index.tsx]: https://github.com/Prop-House/prop-house-monorepo/pull/299/commits/cc47939dcbdbadd70cc962f9e7230063a2311859

**Comment by cryptoseneca** [comment]
Some UX issues I found during testing via desktop: 

- [1] during round ended, weird space below “more”, let’s remove.
- [2] during proposing / voting state, connect to submit prop should be centered
- [3] if you go into a round, click a prop, close the modal and click the back button (to go to the house page), the prop modal will open again
- [4] Use capital letter “No votes allotted”
- [5] once the allotment modal is opened, there’s no way to close it while staying on the same prop



[1]
Screen Shot 2022-12-01 at 12 36 06 PM
[2]
Screen Shot 2022-12-01 at 12 42 15 PM

[3]
Jun 1, 2022 - Jun 8, 2023

[4]
no votes allotted

[5]
have to de

**Comment by cdt-eth** [comment]
### [1] during round ended, weird space below “more”, let’s remove.

### [2] during proposing / voting state, connect to submit prop should be centered
fixed: https://github.com/Prop-House/prop-house-monorepo/pull/299/commits/cc4cebb835c070bcfa14fad47235e3cda81f6854
Screenshot 2022-12-01 at 4 58 50 PM

### [3] if you go into a round, click a prop, close the modal and click the back button (to go to the house page), the prop modal will open again

###[4] Use capital letter “No votes allotted”
https://github.com/Prop-House/prop-house-monorepo/pull/299/commits/cc4cebb835c070bcfa14fad47235e3cda81f6854

### [5] once the allotment modal is opened, there’s no way to close it while staying on the same prop
it closes both on scroll and after clicking the icon again. can you try again? i've verified this on my end.

**Pull request merged**